### ISSUE TITLE
SquaredDifference gives incorrect gradients

### ISSUE THREAD

**Issue by LeeReidCSIRO** [SciSharp/TensorFlow.NET#787]
Hi again,

tf.squared_difference appears to produce incorrect gradients. I don't have a toy example handy, but this is demonstrable by running any training session with an error function like so:

```
Tensor Loss(Tensor y, Tensor expected)
{
return tf.squared_difference(y, expected)/2;
}

```

and comparing to the same run like so:

```
Tensor Loss(Tensor y, Tensor expected)
{
return (y - expected)*(y - expected)/2;
}

```

In my cases, the second works, while the first learns in the wrong direction. Presumably issue is in `nn_grad._SquaredDifferenceGrad` which is just returning the op.inputs and ignoring the `grads` argument, but I don't know how to fix this:

```
        [RegisterGradient("SquaredDifference")]
        public static Tensor[] _SquaredDifferenceGrad(Operation op, Tensor[] grads)
        {
           //"""Returns the gradient for (x-y)^2."""
            Tensor x = op.inputs[0];
            Tensor y = op.inputs[1];
            return new Tensor[]
            {
                x,
                y
            };
        }
```
Thanks

**Comment by Oceania2018** [comment]
@LeeReidCSIRO Could you PR the handy code as UnitTest with `[IgnoreTest]` annotation.
You can write code in `GradientEagerTest`.
![image](https://user-images.githubusercontent.com/1705364/112501305-1fcafd00-8d57-11eb-8580-68d3a53535c5.png)
